Documentation/gettingstarted/kind.rst

```diff
@@ -0,0 +1,244 @@
+      --set global.nodeinit.enabled=true \\
+      --set global.kubeProxyReplacement=partial \\
+      --set global.hostServices.enabled=false \\
+      --set global.externalIPs.enabled=true \\
+      --set global.nodePort.enabled=true \\
+      --set global.hostPort.enabled=true
+
+.. include:: k8s-install-validate.rst
+.. include:: hubble-install.rst
+
+Next steps
+==========
+
+Now that you have a Kubernetes cluster with Cilium up and running, you can take
+a couple of next steps to explore various capabilities:
+
+* :ref:`gs_http`
+* :ref:`gs_dns`
+* :ref:`gs_cassandra`
+* :ref:`gs_kafka`
+
+Tips
+====
+
+Pre-load Cilium images into the kind cluster so each worker doesn't have to pull them.
+
+Run this before :ref:`kind_install_cilium`:
+
+.. parsed-literal::
+
+  docker pull cilium/cilium:|IMAGE_TAG|
+  kind load docker-image cilium/cilium:|IMAGE_TAG|
+
+And add ``--set global.pullPolicy=IfNotPresent`` to Helm install command.
+
+Troubleshooting
+===============
+
+Unable to contact k8s api-server
+--------------------------------
+
+In the :ref:`Cilum agent logs <ts_agent_logs>` you will see::
+
+    level=info msg="Establishing connection to apiserver" host="https://10.96.0.1:443" subsys=k8s
+    level=error msg="Unable to contact k8s api-server" error="Get https://10.96.0.1:443/api/v1/namespaces/kube-system: dial tcp 10.96.0.1:443: connect: no route to host" ipAddr="https://10.96.0.1:443" subsys=k8s
+    level=fatal msg="Unable to initialize Kubernetes subsystem" error="unable to create k8s client: unable to create k8s client: Get https://10.96.0.1:443/api/v1/namespaces/kube-system: dial tcp 10.96.0.1:443: connect: no route to host" subsys=daemon
+
+As Kind is running nodes as containers in Docker, they're sharing your host machines' kernel.
+If :ref:`host-services` wasn't disabled, the eBPF programs attached by Cilium may be out of date
+and no longer routing api-server requests to the current ``kind-control-plane`` container.
+
+Recreating the kind cluster and using the helm command :ref:`kind_install_cilium` will detach the
+inaccurate eBPF programs.
+
+.. _gs_kind_cluster_mesh:
+
+Cluster Mesh
+============
+
+With Kind we can simulate Cluster Mesh in a sandbox too.
+
+Kind Configuration
+------------------
+
+This time we need to create (2) ``config.yaml``, one for each kubernetes cluster.
+We will explicitly configure their ``pod-network-cidr`` and ``service-cidr`` to not overlap.
+
+Example ``kind-cluster1.yaml``:
+
+.. code:: yaml
+
+    kind: Cluster
+    apiVersion: kind.x-k8s.io/v1alpha4
+    nodes:
+    - role: control-plane
+    - role: worker
+    - role: worker
+    - role: worker
+    networking:
+      disableDefaultCNI: true
+      podSubnet: 10.0.0.0/16
+      serviceSubnet: 10.1.0.0/16
+
+Example ``kind-cluster2.yaml``:
+
+.. code:: yaml
+
+    kind: Cluster
+    apiVersion: kind.x-k8s.io/v1alpha4
+    nodes:
+    - role: control-plane
+    - role: worker
+    - role: worker
+    - role: worker
```

Comment:
On my laptop at least, running 3 workers + control plane was a bit slow. Not sure whether this is generally true but we might consider cutting down to 2 workers for these instructions; that should be enough for basic validation of Cilium features, I think?

Reply:
To get all pods deployed by connectivity check to be ready, I had to use 3 workers.